tests/CMakeLists.txt

```diff
@@ -21,6 +21,36 @@ target_include_directories(
         ${CMAKE_CURRENT_SOURCE_DIR}
 )
 
+# Download noc_parameters.h for tests
+set(ARCHS
+    grayskull
+    wormhole
+    blackhole
+)
+foreach(ARCH IN LISTS ARCHS)
+    set(HW_NOC_PARAMETERS_URL
+        "https://raw.githubusercontent.com/tenstorrent/tt-metal/refs/heads/main/tt_metal/hw/inc/${ARCH}/noc/noc_parameters.h"
+    )
+    set(DESTINATION_DIR ${CMAKE_CURRENT_BINARY_DIR}/${ARCH}/noc)
+    file(MAKE_DIRECTORY ${DESTINATION_DIR})
+    get_filename_component(fileName ${HW_NOC_PARAMETERS_URL} NAME)
+    file(
+        DOWNLOAD
```

Comment:
I don't like the idea of introducing a download in consumers.  When it's in tt-metal, can we reference the file from the same source tree (ugly, I know) and guard this with "are we top-level?".

Both options are ugly, but one has a performance cost and a new avenue for failures.

Reply:
I really don't want to do the "are we top level thing".

It disrespects the independence of the two repos. Its also somewhat cyclic to do that.

It could prevent us from using CPM in the future.

Reply:
How do you feel about "only download if the file doesn't exist"?  It'll mean hiccups when needing to consume a newer version of the file, but at least we won't be inflicting papercuts every time CMake reconfigures.

Reply:
sure... on it